### PR TITLE
Update README with newly created OperatorSource

### PR DESCRIPTION
**Pull request by anik120**
Use upstream-community-operators OperatorSource for installation on k8s.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ $ export REGISTRY=<SOME_REGISTRY> \
 ### Description
 The operator manages two CRDs: [OperatorSource](./deploy/crds/operatorsource.crd.yaml) and [CatalogSourceConfig](./deploy/crds/catalogsourceconfig.crd.yaml).
 
-`OperatorSource` is used to define the external datastore we are using to store operator bundles. At the moment we only support Quay's app-registry as our external datastore. Please see [here](deploy/examples/operatorsource.cr.yaml) for an example `OperatorSource`. The `endpoint` in the `spec` is typically set to `https:/quay.io/cnr` if you are using Quay's app-registry. The `registryNamespace` is the name of your app-registry namespace. `displayName` and `publisher` are optional and only needed for UI purposes. If you want an `OperatorSource` to work with private app-registry repositories, please take a look at the [Private Repo Authentication](docs/how-to-auhenticate-private-repositories.md) documentation.
```
Comment (aravindhp) [inline README.md]: You are removing the documentation for private Quay repos. Please add that back.

File: README.md [hunk 2]
```diff
@@ -48,18 +48,18 @@ $ kubectl apply -f deploy/upstream
 ```
 
 #### Installing an operator using Marketplace
-The following section assumes that Marketplace was installed in the `openshift-marketplace` namespace. For Marketplace to function you need to have at least one `OperatorSource` CR present on the cluster. To get started you can use the [community](deploy/examples/operatorsource.cr.yaml) OperatorSource. If you are on an OKD cluster, you can skip this step as it is installed by default.
+The following section assumes that Marketplace was installed in the `openshift-marketplace` namespace. For Marketplace to function you need to have at least one `OperatorSource` CR present on the cluster. To get started you can use the [upstream-community-operators](deploy/examples/operatorsource.cr.yaml) OperatorSource. If you are on an OKD cluster, you can skip this step as it is installed by default.
```
Comment (aravindhp) [inline README.md]: Given you are targeting this for upstream, please update the namespace to `marketplace`. It is better to say `use the OperatorSource` for upstream community operators. Please use `` when referring to `OperatorSource`.

Also update what is installed by default on an OKD cluster.

File: README.md [hunk 3]
```diff
@@ -48,18 +48,18 @@ $ kubectl apply -f deploy/upstream
...
 $ kubectl apply -f deploy/examples/operatorsource.cr.yaml
 ```
 Once the `OperatorSource` has been successfully deployed, you can discover the operators available using the following command:
 ```bash
-$ kubectl get opsrc community-operators -o=custom-columns=NAME:.metadata.name,PACKAGES:.status.packages -n openshift-marketplace
-NAME                  PACKAGES
-community-operators   federationv2,svcat,metering,etcd,prometheus,automationbroker,templateservicebroker,cluster-logging,jaeger,descheduler
+$ kubectl get opsrc upstream-community-operators -o=custom-columns=NAME:.metadata.name,PACKAGES:.status.packages -n openshift-marketplace
+NAME                           PACKAGES
+upstream-community-operators   jaeger
```
Comment (aravindhp) [inline README.md]: Can we show multiple packages here?

File: README.md [hunk 4]
```diff
@@ -48,18 +48,18 @@ $ kubectl apply -f deploy/upstream
...
 ```
 
-Now if you want to install the `descheduler` and `jaeger` operators, create a `CatalogSourceConfig` CR as shown below:
+Now if you want to install the `jaeger` operator, create a `CatalogSourceConfig` CR as shown below:
```
Comment (aravindhp) [inline README.md]: I would like to use this example to call out that you use the same `CatalogSourceConfig` to install multiple operators.

File: deploy/examples/operatorsource.cr.yaml [hunk 5]
```diff
@@ -1,11 +1,11 @@
 apiVersion: marketplace.redhat.com/v1alpha1
 kind: OperatorSource
 metadata:
-  name: community-operators
-  namespace: openshift-marketplace
+  name: upstream-community-operators
+  namespace: marketplace
```
Comment (SamiSousa) [inline deploy/examples/operatorsource.cr.yaml]: Should this be `openshift-marketplace`?

File: deploy/examples/operatorsource.cr.yaml [hunk 6]
```diff
@@ -1,11 +1,11 @@
...
 spec:
   type: appregistry
   endpoint: https://quay.io/cnr
-  registryNamespace: community-operators
-  displayName: "Community Operators"
+  registryNamespace: upstream-community-operators
+  displayName: "Upstream Community Operators"
```
Comment (SamiSousa) [inline deploy/examples/operatorsource.cr.yaml]: The readme inline example CSC doesn't match this display name

File: deploy/examples/operatorsource.cr.yaml [hunk 7]
```diff
@@ -1,11 +1,11 @@
 apiVersion: marketplace.redhat.com/v1alpha1
 kind: OperatorSource
 metadata:
-  name: community-operators
-  namespace: openshift-marketplace
+  name: upstream-community-operators
```
Comment (aravindhp) [inline deploy/examples/operatorsource.cr.yaml]: Please create a new file for this `OperatorSource`.

File: deploy/examples/operatorsource.cr.yaml (hunk 5 above)
Comment (anik120) [reply deploy/examples/operatorsource.cr.yaml]: The section starts with 
> The following section assumes that Marketplace was installed in the openshift-marketplace namespace

I suppose it makes more sense to replace all instances of `openshift-marketplace` with just `marketplace`

File: README.md (hunk 3 above)
Comment (anik120) [reply README.md]: @aravindhp we could once we have multiple operators in the upstream registry. For now we only have jaeger in the registry.

Comment (aravindhp) [reply README.md]: These are just showing examples. It doesn't need to reflect what is exactly there in the repository.

File: README.md (hunk 4 above)
Comment (anik120) [reply README.md]: We don't have descheduler or any other operator besides jaeger in the `upstream-community-operator` OperatorSource. We could update the README to reflect the point above as soon as we have another operator in the `upstream-community-operator` OperatorSource.

Comment (aravindhp) [reply README.md]: Again, we are showing examples here. It doesn't need to reflect exactly what is present in the repository which is going to be changing. We are not going to keep updating the README each time the Quay repo changes.

**Comment by anik120** [comment]
@SamiSousa @aravindhp addressed comments in latest commit. Please review. 

Note: I created a new file called upstream.operatorsource.cr.yaml and renamed the existing operatorsource.cr.yaml to community.operatorsource.cr.yaml

File: README.md [hunk 8]
```diff
@@ -48,43 +48,43 @@ $ kubectl apply -f deploy/upstream
...
-  name: installed-community-operators
-  namespace: openshift-marketplace
+  name: installed-upstream-community-operators
+  namespace: marketplace
 spec:
-  targetNamespace: openshift-operators
-  packages: descheduler,jaeger
-  csDisplayName: "Community Operators"
+  targetNamespace: local-operators
+  packages: descheduler, jaeger
```
Comment (aravindhp) [inline README.md]: No space between the packages

**Comment by anik120** [comment]
@aravindhp Addressed. Also https://github.com/operator-framework/community-operators/pull/86

File: deploy/examples/upstream.operatorsource.cr.yaml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+apiVersion: marketplace.redhat.com/v1alpha1
```
Comment (SamiSousa) [inline deploy/examples/upstream.operatorsource.cr.yaml]: Should we keep both example `OperatorSource` CRs? Having both might suggest that both should be applied, which I don't think we want.

Comment (anik120) [reply deploy/examples/upstream.operatorsource.cr.yaml]: @SamiSousa https://github.com/operator-framework/operator-marketplace/pull/112#discussion_r257819134

I think we decided to keep both examples and direct users to the appropriate example in the appropriate section/discussion.

File: README.md [hunk 10]
```diff
@@ -48,50 +48,51 @@ $ kubectl apply -f deploy/upstream
...
-  namespace: openshift-operators
+  name: jaeger
+  namespace: local-operators
 spec:
   channel: alpha
-  name: descheduler
-  source: installed-community-operators
+  name: jaeger
+  source: installed-upstream-community-operators
+  sourceNamespace: local-operators
```
Comment (SamiSousa) [inline README.md]: :+1:

**Pull request merged**